kong-ngx-build

```diff
@@ -387,6 +387,48 @@ main() {
   succ "Build finished in $SECONDS seconds. Enjoy!"
 }
 
+parse_version() {
+    # in foo-x.y.z-foobar captures x y z
+    local re='[^0-9]*([0-9a-zA-Z]*)(?:[.]([0-9a-zA-Z]*))?(?:[.]([0-9a-zA-Z]*))?.*'
```

Comment:
The last `.*` match and quantifier seem unnecessary.

Reply:
👍 the regex can use some rethinking